### ISSUE TITLE
Some initial remarks

### ISSUE THREAD

**Issue by winks** [hdgarrood/purescript-numeric-hierarchy-guide#1]
Hey,
first a word of caution, apparently I misunderstood "not a mathematician" and I might have had a tad too much math in university (CS major) ;)

But here's some feedback:

a) https://github.com/hdgarrood/purescript-numeric-hierarchy-guide/blame/master/logic.rst#L21

Rereading this sentence for the umpteenth time I cannot regain my original puzzlement when reading it the first three times, but I would rephrase "because P is true" to something like "because at least one of them, in this case P, is true". I first implied some implication, when it's just a simple OR. Maybe a brainfart.

b) https://github.com/hdgarrood/purescript-numeric-hierarchy-guide/blame/master/monoids.rst#L47

You don't define what a "set" is here. Is it in the strict mathematical sense or just "a collection of things"?

c) This is possibly too early a draft, but the "high-school math" references in the intro seem a bit out of place to me. IIRC I learned everything in the Logic chapter in high school (well, the German counterpart) and again in university. Plus there's nothing exciting (yet?) for programmers. Just had to double-check the Monoid chapter with a proper mathematician because I was a bit puzzled that you've explained nothing new to me, just the term Monoid, whereas semigroup is a term I know. (I was told I apparently skipped that one lecture where I should've learned the term :P) Oh, and I am pretty sure I learned the basics of that in high-school as well. TLDR: It's fine, end of wall of text.

In general I think it's nicely written, eager to read more :)

**Comment by hdgarrood** [comment]
> Rereading this sentence for the umpteenth time I cannot regain my original puzzlement when reading it the first three times, but I would rephrase "because P is true" to something like "because at least one of them, in this case P, is true". I first implied some implication, when it's just a simple OR. Maybe a brainfart.

Definitely agree, thanks!

> You don't define what a "set" is here. Is it in the strict mathematical sense or just "a collection of things"?

I am hoping that the difference between these two things won't matter for our purposes; if you are able to point out a way that it does matter, I'd be very keen to know!

> c) This is possibly too early a draft, but the "high-school math" references in the intro seem a bit out of place to me. [...]

Huh, that's interesting, I wasn't aware of that! I certainly wouldn't expect to encounter very much of this material in high-school in the UK. I may reword bits of that, then.

> Plus there's nothing exciting (yet?) for programmers.

Yeah, the important bit (which isn't yet there) is a discussion of rings and fields - I'm not sure if you've used PureScript recently, but the Prelude now defines type classes corresponding to those things.

Thanks very much!

**Comment by winks** [comment]
>> Plus there's nothing exciting (yet?) for programmers.

Was strictly meant for the boolean logic chapter

>  I'm not sure if you've used PureScript recently

I actually gave up while installing due to npm troubles and only read and used http://try.purescript.org a while ago. Still planning to give it a proper try, but JS is not what I'm excited about typically :P

And regarding the high-school stuff: Definitely learnt about the basics of boolean algebra and sets + identity + associativity, just without some of the fancy terms. But it's been a while, no clue how detailed it was.

**Comment by hdgarrood** [comment]
Cool, thanks. I recommend popping in to #purescript IRC if you ever find the time to take another look, there's often people in there who are happy to help :)

I think I've addressed all the things you mentioned in this issue, so I'm going to close it, but feel free to add more comments / open more issues if you think of anything else.
